packages/aws-cdk-lib/aws-codepipeline/lib/trigger.ts

```diff
@@ -32,6 +32,91 @@ export interface GitPushFilter {
   readonly tagsIncludes?: string[];
 }
 
+/**
+ * Git pull request filter for trigger.
+ */
+export interface GitPullRequestFilter {
+  /**
+   * The list of patterns of Git branches that, when pull request events occurs, are
+   * to be excluded from starting the pipeline.
+   *
+   * You can filter with glob patterns. The `branchesExcludes` takes priority
+   * over the `branchesIncludes`.
+   *
+   * Maximum length of this array is 8.
+   *
+   * @default - no branches.
+   */
+  readonly branchesExcludes?: string[];
+
+  /**
+   * The list of patterns of Git branches that, when pull request events occurs, are
+   * to be included as criteria that starts the pipeline.
+   *
+   * You can filter with glob patterns. The `branchesExcludes` takes priority
+   * over the `branchesIncludes`.
+   *
+   * Maximum length of this array is 8.
+   *
+   * @default - no branches.
+   */
+  readonly branchesIncludes?: string[];
+
+  /**
+   * The list of patterns of Git repository file paths that, when pull request events occurs,
+   * are to be excluded from starting the pipeline.
+   *
+   * You can filter with glob patterns. The `filePathsExcludes` takes priority
+   * over the `filePathsIncludes`.
+   *
+   * Maximum length of this array is 8.
+   *
+   * @default - no filePaths.
+   */
+  readonly filePathsExcludes?: string[];
+
+  /**
+   * The list of patterns of Git repository file paths that, when pull request events occurs,
+   * are to be included as criteria that starts the pipeline.
+   *
+   * You can filter with glob patterns. The `filePathsExcludes` takes priority
+   * over the `filePathsIncludes`.
+   *
+   * Maximum length of this array is 8.
+   *
+   * @default - no filePaths.
+   */
+  readonly filePathsIncludes?: string[];
+  
+  /**
+   * The field that specifies which pull request events to filter on (opened, updated, closed)
+   * for the trigger configuration.
+   *
+   * Maximum length of this array is 3.
+   *
+   * @default - no events.
+   */
+  readonly events?: GitPullRequestEvent[];
+}
+
+/**
+ * Event for trigger with pull request filter.
+ */
+export enum GitPullRequestEvent {
+  /**
+   * OPEN
+   */
+  OPEN = 'OPEN',
+  /**
+   * UPDATED
+   */
+  UPDATED = 'UPDATED',
+  /**
+   * CLOSED
+   */
+  CLOSED = 'CLOSED',
+}
```

Comment:
from the API doc.
https://docs.aws.amazon.com/codepipeline/latest/APIReference/API_GitPullRequestFilter.html